docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
+* BookList is always sorted in alphabetical order 
+    * Alternative 1: Sort after adding the book
+        * Pros: Only need to sort after every addition, listing is as per normal
+        * Cons: Adding books may take longer if size of BookList is too large
+    * Alternative 2: Sort before listing the books
+        * Pros: Adding books will not take as long
+        * Cons: Since there are multiple listing methods, may not be the best method to keep sorting before listing.
+
+#### 4.1.2 Find Book by Keyword
+The sequence diagram below demonstrates the command execution process when finding books by a keyword.
+
+![Sequence Diagram for Find Book by Keyword](images/SeqDiagram_FindBook.png)
+
+* Opt-frames ensure that the user input is correct and that the keyword results in a non-empty list of books.
+* Parsing of user input is done in the Parser and FindCommand classes, which is not shown in the diagram.
+* `findByKeyword()` method filters books regardless of case. 
+
+##### Design Considerations
+* Allows user to find books if either title or author contains the keyword.
+    * Pros: Users can find books not just based on book title alone. The search range is increased to author name as well.
+    * Cons: Need to check for both title and author for the list of results.
+* Case insensitive
+    * Pros: Users do not have to bother about the exact letter case when typing the keyword.
+    * Cons: Need to ensure all title or author are converted to the same case before searching.
+* Keyword or phrase
+    * Alternative 1: Let users find by multiple space-separated keywords
+        * Pros: Users can find books if they can only remember part of the book title.
+        * Cons: Search list may not be narrowed as much. 
+    * Alternative 2: Let users find by exact phrase
+        * Pros: Users can narrow down the search using an exact phrase instead of just one word.
+        * Cons: The exact phrase must be typed out for the correct result to show.
+
+### 4.2 Feature: Quote Management
+Given below is the class diagram for classes related to the Quote Management System in Quotesify:
+
+![Class Diagram for Quote Management System](images/ClassDiagram_Quote.png)
+
+* The `XRatingCommand` class represents `AddQuoteCommand`, `ListQuoteCommand`, `EditQuoteCommand`, `DeleteQuoteCommand`,
+`FindQuoteCommand`, `AddQuoteReflectionCommand`, `ListQuoteReflectionCommand`, `EditQuoteReflectionCommand`,
+`DeleteQuoteReflectionCommand`.
+
+#### 4.2.1 Add quote
+The add quote feature allows users to add quotes of multiple formats to Quotesify.
+Quotes added can be of the following format:
+* Quote with author name and reference title
+* Quote with author name
+* Quote with reference title
+* Quote with no author name and reference title
+
+The sequence diagram below reflects the command execution when a user adds a quote to Quotesify.
+
+![Sequence Diagram for Add Quotes](images/SeqDiagram_AddQuote.png)
```

Comment:
![image](https://user-images.githubusercontent.com/7411601/97387126-561f6000-1910-11eb-98be-8ba84c1af692.png)

Since this diagram is _Sequence Diagram_, shall we omit the underline?
Also, to increase legibility, a larger text on the label would be better, right?